js/coinsbit.js

```diff
@@ -0,0 +1,1073 @@
+        market = this.safeMarket (undefined, market);
+        return this.safeTicker ({
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeNumber (tickerFinal, 'high'),
+            'low': this.safeNumber (tickerFinal, 'low'),
+            'bid': this.safeNumber (tickerFinal, 'bid'),
+            'bidVolume': undefined,
+            'ask': this.safeNumber (tickerFinal, 'ask'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': this.safeNumber (tickerFinal, 'open'),
+            'close': last,
+            'last': last,
+            'previousClose': undefined,
+            'change': undefined, // They do have 'change' property, but atm,  for some reason, they always return "0 | -0 | -1 | -2" strange values. So, skipping for now.
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeNumber (tickerFinal, 'volume'),
+            'quoteVolume': this.safeNumber (tickerFinal, 'deal'),
+            'info': ticker,
+        }, market);
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        if (limit !== undefined) {
+            request['limit'] = Math.min (1000, limit); // this parameter is not required, default 50 will be returned
+        }
+        const options = this.safeValue (this.options, 'fetchOrderBook', {});
+        // publicGetDepthResult or publicGetBook
+        const type = this.safeString (params, 'method', 'publicGetDepthResult');
+        const method = this.safeString (options, 'method', type);
+        params = this.omit (params, 'method');
+        let response = undefined;
+        let priceKey = undefined;
+        let amountKey = undefined;
+        if (method === 'publicGetDepthResult') {
+            response = await this[method] (this.extend (request, params));
+            //     {
+            //         asks: [
+            //             [  "46809.06069986", "2.651"  ],
+            //             [  "46819.80796146", "2.6401" ],
+            //             ...
+            //         ],
+            //         bids: [
+            //             [  "46799.39054516",  "2.702" ],
+            //             [  "46801.25196599",  "0.4"   ],
+            //             ...
+            //         ]
+            //     }
+            priceKey = 0;
+            amountKey = 1;
+        } else if (method === 'publicGetBook') {
+            request['offset'] = this.safeString (params, 'offset', 0);
+            // ATM, they need to be called separately (unfortunately)
+            let query = this.extend (request, { 'side': 'buy' });
+            const requestBids = await this[method] (this.extend (query, params));
```

Comment:
I don't think we can have this method that gets the bids and asks separately

Reply:
We could probably add `fetchBidsAsks`, but I'd still keep `fetchOrderBook` and mark it as `'emulated'`.

Reply:
I have seen something like in `btctradeua.js` &  `coinbase ->fetchTicker`  and maybe some other places. 
Personally I, don't think it may be wrong to call 2 requests in one fetch method, as that might not be the axiom that one-fetch should have one request. In opposite, I think, one fetchX unified method should provide same (as much as possible) expected data, and that's what users might be wanting in the end of day from 'fetchX' method. So, if in rare cases (as other ways are unavailable) calling 2 requests in one method, might not be problem , imho - it doesn't break anything at all(?)
But as you say, lmk.

@kroitor this should be 'emulated' because of theoretical difference between the timestamps of the responses right? I see, correct.

Reply:
//btw, but why to mark with emulated, as there I see is second approach 'depth' which gets correct (one request) data for marketdepth. so, it should be still true, and not emulated (emulated can be only the second 'method' type, but in ccxt there is no way to set method partially 'emulated' )

Reply:
In this case setting it to `true` will also work.